scrapy/spidermiddlewares/offsite.py

```diff
@@ -52,7 +52,10 @@ def get_host_regex(self, spider):
         allowed_domains = getattr(spider, 'allowed_domains', None)
         if not allowed_domains:
             return re.compile('') # allow all by default
-        regex = r'^(.*\.)?(%s)$' % '|'.join(re.escape(d) for d in allowed_domains if d is not None)
+        invalid_domains = [d for d in allowed_domains if '/' in d]
+        if invalid_domains:
+            logging.warn("allowed_domains accepts only domains, not URLs. Invalid_domains being ignored:%s." %invalid_domains)
+        regex = r'^(.*\.)?(%s)$' % '|'.join(re.escape(d) for d in allowed_domains if d is not None and d not in invalid_domains)
```

Comment:
hi @gRSsomani , sorry for the late feedback.
I think `if d is not None` could be tested when building `invalid_domains` above.
Also, these lines become quite long. I'd suggest wrapping the line a little.